contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,202 @@
+
+#### Language barriers
+
+A majority of users of VueJs are the Chinese as VueJs is developed by a Chinese American. He is supportive of the Chinese community and hence a lot of the existing plugins are written in Chinese. There might be some language barriers for an English speaking developer seeking for VueJs resources.
+
+## Installation
+
+There are many ways to install VueJs. Some of the following ways are mentioned below:
+* Import VueJs in HTML `<script>` tag:
+  - [development version](https://vuejs.org/js/vue.js)
+  - [product version](https://vuejs.org/js/vue.min.js)
+* CDN:
+  - include the following into the html file:
+    `<script src="https://cdn.jsdelivr.net/npm/vue@2.5.13/dist/vue.js"</script>`
+* NPM:
+```npm
+    $ npm install vue
+```
+* CLI:
+```cli
+    $ npm install --global vue-cli  # install vue-cli
+    $ vue init webpack my-project  # create a new project using the "webpack" template
+    $ cd my-project  # install dependencies and go!
+    $ npm install
+    $ npm run dev
+```
+
+Apart from installing VueJs, you can install [VueJs development tools](https://github.com/vuejs/vue-devtools#vue-devtools) in your browser. This will allow you to inspect and debug the Vue components in your projects.
+
+## Creating a Simple Project in VueJs
+
+Steps to print the VueJs version of "Hello World" (a minimal example to show how easy it is to integrate VueJs into your web project):
+
+* Import VueJs CDN and the JavaScript file in the main HTML file.
+```HTML
+  <script src="https://unpkg.com/vue@2.5.13/dist/vue.js"></script>
+  <script src="the_path_to_the_javacript_file.js"></script>
+```
+
+* Create a Vue instance in the JavaScript file; bind the instance to one of the component in our html file (the `root` element).
+  - In this case, only the `root` component can be accessed in VueJs while the rest are unaffected. This is how we progressively plug in VueJs into our projects.
+```js
+  new Vue ({
+  	el: '#root',
+  });
+```
+```HTML
+  <div id="root"></div>
+```
+
+* Specify our data (message: "Hello World") in the Vue instance.
+```js
+  data: {
+    message: "Hello World"
+  }
+```
+
+* Pass the message to the HTML file using double curly brackets.
+```HTML
+  <div id="root">
+    <h1>{{ message }}</h1>
+  </div>
+```
+
+* Open the brower and we will see "Hello World" being displayed.
+
+Putting all parts together:
+
+```HTML
+<!--This is the main HTML file-->
+<body>
+  <div id="root">
+		<h1>{{ message }}</h1>
+	</div>
+	<script src="https://unpkg.com/vue@2.5.13/dist/vue.js"></script>
+	<script src="the_path_to_the_javacript_file.js"></script>
+</body>
+```
+
+```js
+// This is inside the javacript file
+new Vue ({
+	el: '#root',
+
+  data: {
+    message: "Hello World"
+  }
+});
+```
+
+### Common VueJs syntax
```

Comment:
I think there is no need for this section. It's too much details for someone who is not using this article as a tutorial and too little details for someone who wants to actually use Vue.